File: C#-OOP-Basics/Problems Course Instance 2018/Inheritance/Animals/Cats/Kitten.cs
public class Kitten : Cat
{
    public Kitten(string name, string age) : base(name, age)
    {
        base.Gender = "Female";
    }

    public Kitten(string name, string age, string gender) : this(name, age)
    {
    }

    public override string ProduceSound()
    {
        return "Meow";
    }
}

File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson03Inheritance/Exercise/pr06_animals/Kitten.java
package Lesson03Inheritance.Exercise.pr06_animals;

public class Kitten extends Cat {

    private static final String SOUND = "Miau";

    public Kitten(String name, int age, String gender) {
        super(name, age, gender);
    }

    @Override
    protected void setGender(String gender) {
        if (!MessageConstants.FEMALE.equalsIgnoreCase(gender)) {
            throw new IllegalArgumentException(MessageConstants.INVALID_INPUT);
        }
        super.setGender(gender);
    }

    @Override
    public String produceSound() {
        return SOUND;
    }
}
